snapcraft/internal/parts.py

```diff
@@ -151,5 +174,30 @@ def define(part_name):
               default_flow_style=False, stream=sys.stdout)
 
 
+def search(part_match):
+    header_len = len(_HEADER_PART_NAME)
+    matches, part_length = _RemoteParts().matches_for(part_match, header_len)
+
+    terminal_width = get_terminal_width(max_width=None)
+    part_length = max(part_length, header_len)
+    # <space> + <space> + <description> + ... = 5
+    description_space = terminal_width - part_length - 5
+
+    if not matches:
+        # apt search does not return error, we probably shouldn't either.
+        logger.info('No matches found, try to run `snapcraft update` to '
+                    'refresh the remote parts cache.')
+        return
+
+    print('{}  {}'.format(
+        _HEADER_PART_NAME.ljust(part_length, ' '), _HEADER_DESCRIPTION))
+    for part_key in matches.keys():
+        description = matches[part_key]['description']
+        if len(description) > description_space:
+            description = '{}...'.format(description[0:description_space])
```

Comment:
@kyrofa your idea is not bad, but a bug report would be better so if not `grepping` we present a different layout.